source/core/Ky.ts

```diff
@@ -88,7 +88,8 @@ export class Ky {
 			.finally(async () => {
 				// Now that we know a retry is not needed, close the ReadableStream of the cloned request.
 				if (!ky.request.bodyUsed) {
-					await ky.request.body?.cancel();
+					// Consume stream via .arrayBuffer() to avoid Node.js .cancel() hang bug on cloned streams.
```

Comment:
> This is caused by a long-standing bug in Node.js itself, which still exists in recent versions like v24.7.0. The bug is triggered when await request.body.cancel() is called on a request that has been cloned (for retries) but where neither the original nor the cloned stream has been consumed.

Please include a link to the open Node.js issue here. If there is none, please post one.

Reply:
This should also only be done for Node.js. I don't want to risk some weird behavior in browsers where it's not needed.